pkg/api/types.go

```diff
@@ -2793,6 +2793,11 @@ type ExportOptions struct {
 	Export bool `json:"export"`
 	// Should the export be exact.  Exact export maintains cluster-specific fields like 'Namespace'
 	Exact bool `json:"exact"`
+	// If specified:
+	// - if unset, then the result result is returned from remote storage based on quorum-read flag;
+	// - if it's 0, then we simply return what we currently have in cache, no guarantee;
+	// - if set to non zero, then the result is as fresh as given rv.
+	ResourceVersion string
```

Comment:
I'm fine with internal GetOptions too. I think the main question is what we are going to do in the client. So I would imagine that for all Get() methids in auto-generated clients, e.g.:
https://github.com/kubernetes/kubernetes/blob/master/pkg/client/clientset_generated/release_1_5/typed/core/v1/pod.go#L39
It would take XxxOptions as an argument (similarly to List taking ListOptions). The question is what this Xxx would be (the changes on server side no matter what we decide on will be small).

@lavalamp @smarterclayton - WDYT?